### PR TITLE
Updates so that env loading actually works

### PR DESCRIPTION
**Pull request by janschulz**
A few bugfixes (and quite a rewrite for the lazy loading functionality) + periodic updates in case you create a new environment...

Closes: https://github.com/Cadair/jupyter_environment_kernels/issues/28

### CODE REVIEW

File: environment_kernels/core.py [hunk 1]
```diff
@@ -102,6 +101,17 @@ class EnvironmentKernelSpecManager(KernelSpecManager):
     def __init__(self, *args, **kwargs):
         super(EnvironmentKernelSpecManager, self).__init__(*args, **kwargs)
         self.log.info("Using EnvironmentKernelSpecManager...")
+        try:
+            from tornado.ioloop import PeriodicCallback
+            # every 5 minutes
+            updater = PeriodicCallback(callback=self._update_env_data, callback_time=1000 * 60 * 5)
```
Comment (Cadair) [inline environment_kernels/core.py]: Can we make this time a config variable?

Comment (Cadair) [inline environment_kernels/core.py]: Also it would be really nice if we could hook this into a button in the UI. But that's probably for later.

Comment (janschulz) [inline environment_kernels/core.py]: The config option is no problem, but I've no idea how to add a button in the UI :-(

**Comment by janschulz** [comment]
Added a config item for the refresh interval. I set it to 3 minutes: on my system it takes about 2 seconds to get all (currently 3) envs. I also started a background refresh right in init.

**Comment by janschulz** [comment]
@Cadair: Merge?

**Pull request merged**